Set max_bonds=0 so when bonding driver is loaded, no bonds are created.  Bonds
can be created once driver is loaded using ifenslave (and /etc/network/interfaces).

The kernel default value is max_bonds=1 which creates an empty bond0 on bonding
driver load.

diff --git a/include/linux/if_bonding.h b/include/linux/if_bonding.h
index a17edda..13f3ed0 100644
--- a/include/linux/if_bonding.h
+++ b/include/linux/if_bonding.h
@@ -81,7 +81,7 @@
 #define BOND_STATE_ACTIVE       0   /* link is active */
 #define BOND_STATE_BACKUP       1   /* link is backup */
 
-#define BOND_DEFAULT_MAX_BONDS  1   /* Default maximum number of devices to support */
+#define BOND_DEFAULT_MAX_BONDS  0   /* Default maximum number of devices to support */
 
 #define BOND_DEFAULT_TX_QUEUES 16   /* Default number of tx queues per device */
 
